feat(docs): add messaging page to typescript guide (#1114)
wasmCloud/wasmCloud.com: docs/wash/developer-guide/language-support/typescript/messaging.mdx

## Overview

Messaging typically spans **at least two components**. A component can import `consumer` to publish events without a handler in the same deployment; for example, the component might publish audit events that an external NATS subscriber consumes. A handler can also receive from multiple senders. 

The example template used on this page demonstrates a two-component request-reply pattern that may serve as a starting point. The template consists of:

- **`http-api`** — An HTTP component that accepts a `POST /task` request and uses `wasmcloud:messaging/consumer` to dispatch a request message and await its reply.
- **`task-worker`** — A headless component that exports `wasmcloud:messaging/handler` to receive messages, process them, and publish a reply.

![Messaging flow](../../../images/messaging-flow.svg)

:::info[NATS in production]
`wasmcloud:messaging` uses NATS as its production transport, built into the wasmCloud runtime. During development with `wash dev`, messaging is handled in-process; no NATS server is required. In production, the runtime connects to NATS automatically when the host starts with a NATS URL.
:::

> Adding `wasmcloud:messaging` to a component project touches four files: two WIT worlds and two bundler configs:

Reviewer: ```suggestion
Adding `wasmcloud:messaging` in this component project touches four files: two WIT worlds and two bundler configs:
```

* `http-api/wit/world.wit` — Declares the messaging consumer import in the HTTP component
* `task-worker/wit/world.wit` — Declares the consumer import and handler export
* `http-api/rolldown.config.mjs` — Extends the externals list to cover `wasmcloud:` imports
* `task-worker/rolldown.config.mjs` — Same change as above

Plus the two TypeScript implementation files:

* `http-api/src/component.ts` — Uses `request()` to dispatch tasks and receive replies
* `task-worker/src/component.ts` — Exports a `handler` object with `handleMessage()`

:::tip[Complete example]
A full working example is available as a template:
```bash
wash new https://github.com/wasmCloud/typescript.git \
  --name http-api-with-distributed-workloads \
  --subfolder templates/http-api-with-distributed-workloads
```
:::

## Step 1: Declare the WIT worlds

> Every capability a component uses or provides must be declared in its WIT world. Messaging is split into two interfaces that play different roles.

Reviewer: This sentence doesn't *really* set up the next section (since you jump into per-component WITs)... Maybe a quick list/TOC here would better set up the coming sections?

### HTTP API — import the consumer

The `http-api` component only sends messages; it never receives them. In `http-api/wit/world.wit`, declare a single import:

```wit {4}
package wasmcloud:templates@0.1.0;

world typescript-http-api-with-distributed-workloads-api {
  import wasmcloud:messaging/consumer@0.2.0;

  export wasi:http/incoming-handler@0.2.6;
}
```

### Task worker — import consumer, export handler

The `task-worker` component receives messages **and** sends replies. Receiving requires exporting `handler`; replying requires importing `consumer`. In `task-worker/wit/world.wit`:

```wit {4-5}
package wasmcloud:templates@0.1.0;

world typescript-http-api-with-distributed-workloads-worker {
  import wasmcloud:messaging/consumer@0.2.0;

  export wasmcloud:messaging/handler@0.2.0;
}
```

### What these interfaces provide

| Interface | Direction | Purpose |
|---|---|---|
| `wasmcloud:messaging/consumer` | import | Send messages (`request`, `publish`) |

> | `wasmcloud:messaging/handler` | export | Receive messages (`handleMessage`) |

Reviewer: A link to these interfaces' definitions might be reasonable here so people can peruse the WIT themselves?

The underlying message type used by both interfaces:

```wit
record broker-message {
  subject: string,
  body: list<u8>,       // Uint8Array in TypeScript
  reply-to: option<string>,
}
```

### How dependency resolution works

When you run `npm run build`, the build pipeline calls `wash wit fetch` as a setup step. `wasmcloud:messaging` is hosted at the `wasmcloud.com` package registry — distinct from the `wasi.dev` registry used for standard WASI interfaces. Both registries are resolved automatically; no extra configuration is required.

## Step 2: Update `rolldown.config.mjs`

Unlike `wasi:` imports, `wasmcloud:` imports are not covered by the default external pattern in `rolldown.config.mjs`. Both components need this pattern extended.

**Before** (default in single-component templates):

> ```javascript

Reviewer: We can probably use the `diff` codeblock syntax here? not sure if you can combine it with another language so we can get both kinds of highlighting but at the very least diff would get the point across I think!
